modules/gcp_nodepool/gcp_nodepool.py

```diff
@@ -0,0 +1,47 @@
+from typing import TYPE_CHECKING
+
+from modules.base import ModuleProcessor
+from opta.core.gcp import GCP
+from opta.exceptions import UserErrors
+
+if TYPE_CHECKING:
+    from opta.layer import Layer
+    from opta.module import Module
+
+
+class GcpNodePoolProcessor(ModuleProcessor):
+    def __init__(self, module: "Module", layer: "Layer"):
+        super(GcpNodePoolProcessor, self).__init__(module, layer)
+
+    def process(self, module_idx: int) -> None:
+        if self.module.data.get("guest_accelerator_count", 0):
+            gcp_gke_module = None
+            for module in self.layer.modules:
+                if (module.aliased_type or module.type) == "gcp-gke":
+                    gcp_gke_module = module
+                    break
+            if gcp_gke_module is None:
+                raise UserErrors(
+                    "The gcp-nodepool module needs to be run on the same yaml as gcp-gke"
+                )
+            gcp_gke_node_zone_names = set(gcp_gke_module.data["node_zone_names"])
+            # Identify the intersection of the GKE cluster's node locations, and zones
+            # where the desired guest accelerators are available.
+            gcp_nodepool_node_zone_names = sorted(
```

Comment:
I would suggest not shrinking the configured node zone list and instead raise an error and make the user act. Otherwise the materialized tf is non-deterministic and misleading

Reply:
Perhaps I am misunderstanding the comment. The list of zones where GPUs are available can be a subset of the zones where the control plane is running. We wouldn't want to run the control plane on fewer zones just because GPUs are not available on all zones. Also this node zone list is auto-populated by the `get_current_zones` [method](https://github.com/unionai/opta/blob/8fc65081c3a85091c941985a32e2ac8cbf6511e7/opta/core/gcp.py#L135) right now, and isn't user-facing. See:
https://github.com/unionai/opta/blob/8fc65081c3a85091c941985a32e2ac8cbf6511e7/modules/gcp_gke/gcp_gke.py#L40
https://github.com/unionai/opta/blob/8fc65081c3a85091c941985a32e2ac8cbf6511e7/modules/gcp_gke/gcp-gke.yaml#L22
https://github.com/unionai/opta/blob/8fc65081c3a85091c941985a32e2ac8cbf6511e7/modules/gcp_nodepool/gcp-nodepool.yaml#L17

Reply:
Sorry, my mistake. Initially interpreted this as a user-facing var and forgot to confirm/reconcile my comment. Agree on your commentary - thanks for clarifying